experiments.py

```diff
@@ -0,0 +1,162 @@
+                    train_timesteps=TRAIN_TIMESTEPS, 
+                    board_height=BOARD_HEIGHT, 
+                    board_width=BOARD_WIDTH, 
+                    visualization_fps=VIS_FPS, 
+                    reward_function=RewardFuncs.reward_closer_to_fruit)
+    scores = testRL(model=model, 
+                    test_timesteps=TEST_TIMESTEPS, 
+                    board_height=BOARD_HEIGHT, 
+                    board_width=BOARD_WIDTH, 
+                    visualize_testing=VISUALIZE_TESTING, 
+                    visualization_fps=VIS_FPS, 
+                    reward_function=RewardFuncs.reward_closer_to_fruit)
+    analyze_and_write_to_csv(strategy_label, strategy_description, scores)
+
+    # Kill after 10 idle moves
+    strategy_label = "("+model_type+"): "+"Kill after 10 idle moves"
+    strategy_description = "Here we just do the basic reward structure of + for fruit and - for wall. We kill the snake after a set number of idle moves."
+    model = trainRL(model_generator=model_generator,
+                    train_timesteps=TRAIN_TIMESTEPS, 
+                    board_height=BOARD_HEIGHT, 
+                    board_width=BOARD_WIDTH,
+                    max_moves_no_fruit=10,
+                    visualization_fps=VIS_FPS, 
+                    reward_function=RewardFuncs.basic_reward_func)
+    scores = testRL(model=model, 
+                    test_timesteps=TEST_TIMESTEPS, 
+                    board_height=BOARD_HEIGHT, 
+                    board_width=BOARD_WIDTH, 
+                    max_moves_no_fruit=10,
+                    visualize_testing=VISUALIZE_TESTING, 
+                    visualization_fps=VIS_FPS, 
+                    reward_function=RewardFuncs.basic_reward_func)
+    analyze_and_write_to_csv(strategy_label, strategy_description, scores)
+
+    # Kill after 30 idle moves
+    strategy_label = "("+model_type+"): "+"Kill after 30 idle moves"
+    strategy_description = "Here we just do the basic reward structure of + for fruit and - for wall. We kill the snake after a set number of idle moves."
+    model = trainRL(model_generator=model_generator,
+                    train_timesteps=TRAIN_TIMESTEPS, 
+                    board_height=BOARD_HEIGHT, 
+                    board_width=BOARD_WIDTH,
+                    max_moves_no_fruit=30,
+                    visualization_fps=VIS_FPS, 
+                    reward_function=RewardFuncs.basic_reward_func)
+    scores = testRL(model=model, 
+                    test_timesteps=TEST_TIMESTEPS, 
+                    board_height=BOARD_HEIGHT, 
+                    board_width=BOARD_WIDTH, 
+                    max_moves_no_fruit=30,
+                    visualize_testing=VISUALIZE_TESTING, 
+                    visualization_fps=VIS_FPS, 
+                    reward_function=RewardFuncs.basic_reward_func)
+    analyze_and_write_to_csv(strategy_label, strategy_description, scores)
+
+    # Kill and Punish after 10 idle moves
```

Comment:
I added in the following experiments:
* Kill after N idle moves
* Kill and Punish after N idle moves

Is there a way to do "Punish after N idle moves" where the snake gets punished but not killed?
@davidd-55 @dave-carroll7

Reply:
I'm able to do this per move (i.e. every move that no collision or fruit occurs -0.5, but can't seem to just do this after N moves. Although I don't know if this is desirable

Reply:
Pretty sure all we would need to do to accomplish "Punish after N idle moves" is remove `did_exceed_max_moves` from the conditional that determines `done` in `snake_env.py`, that way the reward function would punish for N idle moves but it would not tell the RL to restart based on that condition.